compat/src/index.d.ts

```diff
@@ -68,12 +68,858 @@ declare namespace React {
+		src?: string | undefined;
+		srcSet?: string | undefined;
+		type?: string | undefined;
+		width?: number | string | undefined;
+	}
+
+	interface StyleHTMLAttributes<T> extends HTMLAttributes<T> {
+		media?: string | undefined;
+		scoped?: boolean | undefined;
+		type?: string | undefined;
+	}
+
+	interface TableHTMLAttributes<T> extends HTMLAttributes<T> {
+		cellPadding?: number | string | undefined;
+		cellSpacing?: number | string | undefined;
+		summary?: string | undefined;
+		width?: number | string | undefined;
+	}
+
+	interface TextareaHTMLAttributes<T> extends HTMLAttributes<T> {
+		autoComplete?: string | undefined;
+		cols?: number | undefined;
+		dirName?: string | undefined;
+		disabled?: boolean | undefined;
+		form?: string | undefined;
+		maxLength?: number | undefined;
+		minLength?: number | undefined;
+		name?: string | undefined;
+		placeholder?: string | undefined;
+		readOnly?: boolean | undefined;
+		required?: boolean | undefined;
+		rows?: number | undefined;
+		value?: string | ReadonlyArray<string> | number | undefined;
+		wrap?: string | undefined;
+
+		onChange?: ChangeEventHandler<T> | undefined;
+	}
+
+	interface TdHTMLAttributes<T> extends HTMLAttributes<T> {
+		align?: 'left' | 'center' | 'right' | 'justify' | 'char' | undefined;
+		colSpan?: number | undefined;
+		headers?: string | undefined;
+		rowSpan?: number | undefined;
+		scope?: string | undefined;
+		abbr?: string | undefined;
+		height?: number | string | undefined;
+		width?: number | string | undefined;
+		valign?: 'top' | 'middle' | 'bottom' | 'baseline' | undefined;
+	}
+
+	interface ThHTMLAttributes<T> extends HTMLAttributes<T> {
+		align?: 'left' | 'center' | 'right' | 'justify' | 'char' | undefined;
+		colSpan?: number | undefined;
+		headers?: string | undefined;
+		rowSpan?: number | undefined;
+		scope?: string | undefined;
+		abbr?: string | undefined;
+	}
+
+	interface TimeHTMLAttributes<T> extends HTMLAttributes<T> {
+		dateTime?: string | undefined;
+	}
+
+	interface TrackHTMLAttributes<T> extends HTMLAttributes<T> {
+		default?: boolean | undefined;
+		kind?: string | undefined;
+		label?: string | undefined;
+		src?: string | undefined;
+		srcLang?: string | undefined;
+	}
+
+	interface VideoHTMLAttributes<T> extends MediaHTMLAttributes<T> {
+		height?: number | string | undefined;
+		playsInline?: boolean | undefined;
+		poster?: string | undefined;
+		width?: number | string | undefined;
+		disablePictureInPicture?: boolean | undefined;
+		disableRemotePlayback?: boolean | undefined;
+	}
+
+	// this list is "complete" in that it contains every SVG attribute
+	// that React supports, but the types can be improved.
+	// Full list here: https://facebook.github.io/react/docs/dom-elements.html
+	//
+	// The three broad type categories are (in order of restrictiveness):
+	//   - "number | string"
+	//   - "string"
+	//   - union of string literals
+	interface SVGAttributes<T> extends AriaAttributes, DOMAttributes<T> {
```

Comment:
Any reason why we're not simply reexporting Preact's types here?

https://github.com/preactjs/preact/blob/7cd48335e709f0b32d83070ae320f2010943c310/src/jsx.d.ts#L79

Reply:
We want to avoid duplication as much as possible, copying huge segments from React's types is probably undesirable

Reply:
Thanks for reviewing and comments. 
I updated the PR and removed the duplicated code and checked others.

Reply:
Sorry, I wasn't saying that they should be removed outright, but that we should be re-exporting core's definitions where at all possible.

Most of these are on the JSX namespace in core and therefore do need to be reexported if users want to import the type directly.

Reply:
Hmm, at first i thought this types are from `IntrinsicElements`. but they not. These types generated by react. 

For example we have `HTMLButtonElement` that comes from mdn, but here we have a new interface of `ButtonHTMLAttributes`

So i don't know is it possible to generate all them by a simple code?